DelvUI/Interface/StatusEffects/StatusEffectsList.cs

```diff
@@ -126,15 +135,34 @@ private List<StatusEffectData> StatusEffectsData()
                 list.Add(new StatusEffectData(status, row));
             }
 
-            return list;
+            if (filterBuffs.Count == 0) {
+                return list;
+            }
+
+            // Always adhere to the priority of buffs set by filterBuffs.
+            var toReturn = new List<StatusEffectData>();
+            foreach (uint buffId in filterBuffs)
+            {
+                var idx = list.FindIndex(s => (uint)s.statusEffect.EffectId == buffId);
+                if (idx >= 0)
+                {
+                    toReturn.Add(list[idx]);
+                }
+            }
+            return toReturn;
```

Comment:
Instead of iterating this after going trough the actor buffs, maybe it's better to do it directly in the for..loop right before.
`if (filterBuffs.Contains(status,EffectId) continue;`

Reply:
If you put it in the above loop, you can't guarantee that the resulting buff array follows the same priority as what's in filter buffs. 

Since this loop isn't nested it shouldn't affect the actual overall complexity of the function.

Reply:
You're totally right, didn't realize the filter buffs were also used to sort the list. Makes sene!